packages/next/src/next-request-context.ts

```diff
@@ -0,0 +1,62 @@
+/*
+WARNING: WATCH OUT IF YOU'RE MODIFYING THIS FILE!
+The types here must be kept in sync (or at least backwards-compatible)
+with "@next/request-context" in the Next.js codebase.
+
+This module defines `globalThis[Symbol.for("@next/request-context")]`
+which Next.js uses to access `waitUntil`.
+It is expected that the platform provides this.
+*/
+
+// TODO: why is this failing?
+//   Could not resolve "async_hooks" (use "platform: 'node'" when building for node)
+import { AsyncLocalStorage } from 'async_hooks';
+
+const name = '@next/request-context';
+export const NEXT_REQUEST_CONTEXT_SYMBOL = Symbol.for(name);
+
+const INTERNAL_STORAGE_FIELD_SYMBOL = Symbol.for('internal.storage');
+
+export type NextRequestContextValue = {
+  waitUntil?: WaitUntil;
+};
+
+export type NextRequestContext = {
+  get(): NextRequestContextValue;
+  /** @ignore */
+  [INTERNAL_STORAGE_FIELD_SYMBOL]: AsyncLocalStorage<NextRequestContextValue>;
+};
```

Comment:
unclear if this is the shape we want to go with (vs just exposing a vanilla AsyncLocalStorage directly)

i'd argue that it might be useful to not expose a full ALS here, because
- we don't need (or want) the full power of ALS in next.js, we're just reading it
- this contract could technically be fulfilled using [AsyncContext](https://github.com/tc39/proposal-async-context), so coupling ourselves to an AsyncLocalStorage could prove annoying in the future
  - for single-threaded invocations (e.g. a lambda that only processes one request at a time) , it could even be a plain global that gets reset between invocations!
  
 cc @wyattjoh